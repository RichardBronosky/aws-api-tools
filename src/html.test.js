import React from 'react'
import renderer from 'react-test-renderer'
import HTML from './html'

describe('HTML', () => {
  const testProps = {
    htmlAttributes: { test1: 'html_attribute' },
    headComponents: [<div key="head_comps">test head comp</div>],
    bodyAttributes: { test2: 'body_attribute' },
    preBodyComponents: [<div key="pre_body">test pre pody</div>],
    body: 'test_body',
    postBodyComponents: [<div key="post_body">test post pody</div>],
  }

  it('renders correctly', () => {
    const tree = renderer.create(<HTML { ...testProps } />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
