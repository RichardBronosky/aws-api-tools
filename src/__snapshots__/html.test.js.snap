// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HTML renders correctly 1`] = `
<html
  lang="en"
  test1="html_attribute"
>
  <head>
    <meta
      charSet="utf-8"
    />
    <meta
      content="ie=edge"
      httpEquiv="x-ua-compatible"
    />
    <meta
      content="width=device-width, initial-scale=1, shrink-to-fit=no"
      name="viewport"
    />
    <div>
      test head comp
    </div>
  </head>
  <body
    test2="body_attribute"
  >
    <div>
      test pre pody
    </div>
    <div
      aria-atomic="true"
      aria-live="assertive"
      aria-relevant="additions"
      className="hide508"
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
      id="aria-announce"
      role="status"
    />
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "test_body",
        }
      }
      id="___gatsby"
    />
    <div>
      test post pody
    </div>
  </body>
</html>
`;
